src/drtsans/filter_events.py

```diff
@@ -0,0 +1,412 @@
+from operator import itemgetter
+from typing import List, Optional, Tuple
+
+from mantid.api import AnalysisDataService, WorkspaceGroup
+from mantid.dataobjects import EventWorkspace
+from mantid.simpleapi import (
+    AddSampleLog,
+    CreateEmptyTableWorkspace,
+    FilterEvents,
+    GenerateEventsFilter,
+    GroupWorkspaces,
+    LoadEventNexus,
+    logger,
+    mtd,
+)
+
+from drtsans import PV_POLARIZER_FLIPPER, PV_ANALYZER_FLIPPER
+from drtsans.polarization import PV_POLARIZER, PV_POLARIZER_VETO, PV_ANALYZER, PV_ANALYZER_VETO
+from drtsans.samplelogs import SampleLogs
+from drtsans.type_hints import MantidWorkspace
+
+
+def workspace_handle(input_workspace: MantidWorkspace):
```

Comment:
this is a very general function. How about you move it to module dataobjects.py? To me it seems more related to the purpose of the module